chore(dot/parachain): consistently use ParaID type

haikoschol opened the pull request:
## Changes

No functional changes, just mechanically replacing `uint32` with `parachaintypes.ParaID`.

## Tests

```sh
go test ./...
```

## Issues

Closes #3883

haikoschol committed:
chore(dot/parachain): consistently use ParaID type

axaysagathiya commented:
TestCandidateValidation_processMessageValidateFromChainState test fails because of `Unexpected call to *candidatevalidation.MockInstance.ParachainHostPersistedValidationData`.

We will have to modify the test to include the mock instance as one of the test-case fields. Each test case will have its own mock instance, so we can run all test cases in parallel.

haikoschol committed:
fix(dot/parachain): candidate validation tests

Some changes to parachaintypes.ParaID were missing in mocked calls,
which let to the tests not terminating in GitHub Actions.

The fix slightly changes the semantics of the tests in
TestCandidateValidation_processMessageValidateFromChainState. Before,
one candidate validation subsystem was created and used for all
subtests. Now one is created for each subtest.
